GitUI/CommandsDialogs/FormDeleteRemoteBranch.cs

```diff
@@ -0,0 +1,119 @@
+
+        private readonly string _defaultRemoteBranch;
+        private readonly HashSet<string> mergedBranches = new HashSet<string>();
+
+        public FormDeleteRemoteBranch(GitUICommands aCommands, string defaultRemoteBranch)
+            : base(aCommands)
+        {
+            InitializeComponent();
+            Translate();
+            _defaultRemoteBranch = defaultRemoteBranch;
+        }
+
+        private void FormDeleteRemoteBranchLoad(object sender, EventArgs e)
+        {
+            Branches.BranchesToSelect = Module.GetRefs(tags: true, branches: true).Where(h => h.IsRemote).ToList();
+            foreach (var branch in Module.GetMergedRemoteBranches())
+            {
+                mergedBranches.Add(branch);
+            }
+
+            if (_defaultRemoteBranch != null)
+            {
+                Branches.SetSelectedText(_defaultRemoteBranch);
+            }
+        }
+
+        private void OkClick(object sender, EventArgs e)
+        {
+            try
+            {
+                if (!DeleteRemote.Checked)
+                {
+                    return;
+                }
+
+                List<IGitRef> selectedBranches = Branches.GetSelectedBranches().ToList();
+
+                var hasUnmergedBranches = selectedBranches.Any(branch => !mergedBranches.Contains(branch.CompleteName));
+
+                if (hasUnmergedBranches)
+                {
+                    if (MessageBox.Show(this, _confirmDeleteUnmergedRemoteBranchMessage.Text, _deleteRemoteBranchesCaption.Text, MessageBoxButtons.YesNo) != DialogResult.Yes)
+                    {
+                        return;
+                    }
+                }
+
+                foreach (var remoteGroup in selectedBranches.GroupBy(b => b.Remote))
+                {
+                    string remote = remoteGroup.Key;
+
+                    EnsurePageant(remote);
+
+                    var cmd = new GitDeleteRemoteBranchesCmd(remote, remoteGroup);
+
+                    ScriptManager.RunEventScripts(this, ScriptEvent.BeforePush);
+
+                    using (var form = new FormRemoteProcess(Module, cmd.ToLine())
+                    {
+                        Remote = remote
+                    })
+                    {
+                        form.ShowDialog();
+
+                        if (!Module.InTheMiddleOfAction() && !form.ErrorOccurred())
+                        {
+                            ScriptManager.RunEventScripts(this, ScriptEvent.AfterPush);
+                        }
+                    }
+                }
+
+                UICommands.RepoChangedNotifier.Notify();
+            }
+            catch (Exception ex)
+            {
+                Trace.WriteLine(ex.Message);
+            }
+            Close();
+        }
+
+        private void DeleteRemote_CheckedChanged(object sender, EventArgs e)
+        {
+            Delete.Enabled = ((CheckBox)sender).Checked;
```

Comment:
please change it to `Delete.Enabled = DeleteRemote.Checked;` this way there is no unsafe cast.